apps/super-app/src/services/CmsSmokeTest.ts

```diff
@@ -0,0 +1,30 @@
+const smokeTestEndpoint = '/api/smoke-test';
+
+// TODO move this into a better module
+// Strapi wraps each model in a "data". Inside which, the things we care about
+// are in "attributes".
+interface StrapiModelResponse<T> {
+  data: {
+    attributes: T;
+  };
+}
+
+interface SmokeTestView {
+  test: boolean;
+}
+
+async function isUp(): Promise<boolean> {
+  const response = await fetch(smokeTestEndpoint);
```

Comment:
maybe it'd be nice to have a way to handle an error here, if the strapi API endpoint is down? i'm getting response status 500 with the error message `Proxy error: Could not proxy request /api/smoke-test from localhost:3000 to http://localhost:1337 (ECONNREFUSED)` locally when strapi dev is not up

but then again this could be good enough. i'm also not familiar with smoke tests and don't know how important it is for this code to be robust, is this test meant to be permanent?

Reply:
This is meant to be a rudimentary simulation of making a call to Strapi to get content (looks like I need to document that better). I'm not sure if there's a good way to handle errors at this level; we _want_ to report that something went wrong so that we can surface it on the UI. We could make this never throw, but for better or worse (probably worse, I hate exception throwing as a pattern), throwing exceptions is an widespread pattern in JS (e.g. it has first class support in React and allows us to display an "oops something went wrong" if we want, like in [`personal-calls/`](https://github.com/bettersg/call-home/blob/5fdc009020fb058da2c40d9f6f0bcffae5af1a11/personal-calls/frontend/src/App.tsx).

I should document that this isn't meant to be permanent; I don't foresee a smoke test like this being useful unless we also add additional telemetry/error reporting e.g. Sentry.